### PR TITLE
added Virtuoso detection

### PR DESCRIPTION
**Pull request by seralf**
added Virtuoso detection, as requested in #1516

### CODE REVIEW

**Comment by AliasIO** [comment]
There is no "Server" property, you probably meant to use "headers". Also you're using a backreference for the version number without using a capture group.

**Comment by seralf** [comment]
hi thanks for the suggestion. I've updated the configuration for Virtuoso, is it correct now? (I still see travis failed but never understand why it's failing)

**Comment by seralf** [comment]
wow, I think I have collected far more errors than anyone else, but it finally seems to be ok :-)

File: src/apps.json [hunk 1]
```diff
@@ -8433,6 +8433,20 @@
...
+		"Virtuoso": {
+			"cats": [
+				"34"
+			],
+			"website": "https://virtuoso.openlinksw.com/",
+			"meta": {
+				"Keywords": "^OpenLink Virtuoso Sparql",
+				"Copyright": "^Copyright &copy; \\d{4} OpenLink Software"
+			},
+			"url": "(?:)[^\/]+\/sparql\\;confidence:20",
```
Comment (gadcam) [inline src/apps.json]: This pattern isn't relevant as it will trigger a lot of false positive IMO. What do you think @AliasIO  ?

**Comment by seralf** [comment]
what pattern? do you mean the url?
Is relevant as we expect almost every virtuoso installtion to have a
/sparql path, but it can be found for every triplestore as well. My idea
was to provide an enforcement to other rules, if you think it is not needed
please remove it: I have no way to test it directly

2017-05-19 18:41 GMT+02:00 Camille Barneaud <notifications@github.com>:

**Comment by gadcam** [comment]
If you open the pull request in github you will see I commented under the URL pattern ;)
I'm only a contributor so I can't remove anything and take my comments for what they are : advises.
IMO you don't need this pattern as it will trigger more false positive than good detections and you already found reliables ways to detect Virtuoso.

File: src/apps.json (hunk 1 above)
Comment (AliasIO) [inline src/apps.json]: Agree, it should be removed or made more specific.

**Comment by seralf** [comment]
ok, I added all the suggested modifications

**Pull request merged**